### PR TITLE
add query toMap function

### PR DESCRIPTION
**Pull request by keiSunagawa**
`to[F]` do can't convert to `Map[K, V]`.
this PR added function `toMap` , that can only call if type `Query0[(K, V)]`

### CODE REVIEW

File: modules/core/src/test/scala/doobie/util/query.scala [hunk 1]
```diff
@@ -23,11 +23,15 @@ class queryspec extends Specification {
...
 
   val q = Query[String,Int]("select 123 where ? = 'foo'", None)
+  val pairQuery = Query[String, (String, Int)]("select 'xxx', 123 where ? = 'foo'", None)
 
   "Query (non-empty)" >> {
     "to" in {
       q.to[List]("foo").transact(xa).unsafeRunSync must_=== List(123)
     }
+    "toMap" in {
+      pairQuery.toMap[String, Int]("foo").transact(xa).unsafeRunSync must_=== Map("xxx" -> 123)
```
Comment (keiSunagawa) [inline modules/core/src/test/scala/doobie/util/query.scala]: type annotation not needs if only scala 2.13.

**Comment by tpolecat** [comment]
I keep hoping there's some way to do this by generalizing `.to` but this seems fine. Will need to do another merge but you're next in line.

**Comment by keiSunagawa** [comment]
I did merge  succeed, I think.
please check :bow: .

**Comment by keiSunagawa** [comment]
@tpolecat 
Sorry. I may not be able grasp your intention.
> I keep hoping there's some way to do this by generalizing .to but this seems fine. Will need to do another merge but you're next in line.

I think "another merge" for resolved, because I see "All checks have passed" on Github web ui.
but, this PR was not merge in this 1 week.
If I'm misunderstanding, you able to close this PR. 🙏

**Comment by tpolecat** [comment]
Thanks!

**Pull request merged**